Detectors/TRD/calibration/src/PadCalibCCDBBuilder.cxx

```diff
@@ -0,0 +1,551 @@
+    int sizeOfHotArea = isolatedHotPadsContainmentSize(hDet, coordinates[2], coordinates[3]);
+    if (sizeOfHotArea > 0 && sizeOfHotArea < areaContainedWithin) {
+      replaceIsolatedHotPads(hDet, coordinates[2], coordinates[3], sizeOfHotArea);
+    }
+  }
+}
+
+void PadCalibCCDBBuilder::checkIfSmallerCloserToCenter(TH2F* hDet, std::vector<int> coordinates, float allowedDifference)
+{
+  if (hDet->GetBinContent(coordinates[0], coordinates[1]) == 0 || hDet->GetBinContent(coordinates[2], coordinates[3]) == 0)
+    return;
+
+  float xCenter = hDet->GetNbinsX() / 2.;
+  float yCenter = hDet->GetNbinsY() / 2.;
+
+  std::vector<float> vCenter, vPad1, vPad2;
+  vCenter.push_back(hDet->GetNbinsX() / 2.);
+  vCenter.push_back(hDet->GetNbinsY() / 2.);
+
+  vPad1.push_back(hDet->GetXaxis()->GetBinCenter(coordinates[0]));
+  vPad1.push_back(hDet->GetYaxis()->GetBinCenter(coordinates[1]));
+
+  vPad2.push_back(hDet->GetXaxis()->GetBinCenter(coordinates[2]));
+  vPad2.push_back(hDet->GetYaxis()->GetBinCenter(coordinates[3]));
+
+  float dist1 = computeDistance(vPad1, vCenter);
+  float dist2 = computeDistance(vPad2, vCenter);
+
+  if ((dist1 < dist2) && ((hDet->GetBinContent(coordinates[2], coordinates[3]) - hDet->GetBinContent(coordinates[0], coordinates[1])) > allowedDifference)) {
+    replacePadCloserToCenter(hDet, coordinates[0], coordinates[1]);
+  }
+
+  if ((dist1 > dist2) && ((hDet->GetBinContent(coordinates[0], coordinates[1]) - hDet->GetBinContent(coordinates[2], coordinates[3])) > allowedDifference)) {
+    replacePadCloserToCenter(hDet, coordinates[2], coordinates[3]);
+  }
+}
+
+std::vector<int> PadCalibCCDBBuilder::compareGain(TH2F* hDet, int column, int row, int shiftcolumn, int shiftrow, float allowedDifference)
+{
+  std::vector<int> coordinates = {-1, -1, -1, -1};
+
+  int colMax = hDet->GetNbinsX();
+  int rowMax = hDet->GetNbinsY();
+
+  // only allow shift along one axis and only by one pad
+  if (!(shiftcolumn == 1 && shiftrow == 0) && !(shiftcolumn == 0 && shiftrow == 1))
+    return coordinates;
+
+  // cheks that the pad is valid
+  if ((column >= colMax) || (row >= rowMax))
+    return coordinates;
+
+  if ((column == colMax && shiftcolumn == 1) || (row == rowMax && shiftrow == 1))
+    return coordinates; // do not compare with overflow
+
+  float gain1 = hDet->GetBinContent(column, row);
+  float gain2 = hDet->GetBinContent(column + shiftcolumn, row + shiftrow);
+
+  if ((gain1 == 0 && gain2 > 0) || (gain1 > 0 && gain2 == 0) || (abs(gain1 - gain2) > allowedDifference)) {
+    coordinates[0] = column;
+    coordinates[1] = row;
+    coordinates[2] = column + shiftcolumn;
+    coordinates[3] = row + shiftrow;
+  }
+
+  return coordinates;
+}
+
+float PadCalibCCDBBuilder::computeDetectorAverage(TH2F* hDet)
+{ // computes an average over filled cells
+  // cells are accessed through their values, not bin coordinates
+  // the average os computed over absolute values as inter/extrapolated cells
+  // have negatiove values
```

Comment:
I don't think that doxygen pick this up as a valid format to build docs? @tdietel @martenole 
I think you should move these above the functions.